packages/node_modules/pouchdb-adapter-indexeddb/src/rewrite.js

```diff
@@ -77,8 +77,8 @@ function needsRewrite(data) {
       return true;
     } else if (data[key] === null || typeof data[key] === 'boolean') {
       return true;
-    } else if (typeof data[key] === 'object') {
-      return needsRewrite(data[key]);
+    } else if (typeof data[key] === 'object' && needsRewrite(data[key])) {
+      return true;
```

Comment:
Did this need to change?  It's not immediately obvious why it would need to.

Reply:
The problem here is that the return cancels the for loop. That means when `data[key]` is an object the `needsRewrite` will be evaluated for that object, but if there are any more keys in the outer object they will be ignoring, leading to false negatives.

Reply:
:bulb: